### ISSUE TITLE
button group accessibility - Ensures every form element has a label

### ISSUE THREAD

**Issue by Lausselloic** [twbs/bootstrap#22302]
https://v4-alpha.getbootstrap.com/components/button-group/#button-toolbar

Ensures every form element has a label, the two sample for mail adress doesn't have a valid label (aria-described is not enough). I suggest to add an aria-label, which will reuse the palceholder content

e.g:
```
<span class="input-group-addon" id="btnGroupAddon">@</span>
<input type="text" class="form-control" placeholder="Input group example" aria-describedby="btnGroupAddon" aria-label="Input group example">
```

**Comment by Lausselloic** [comment]
Maybe need also an update for all form elements also on input-group page. And need to update the accessibility best practice : aria-describedby and placeholder are NOT a valid technique to provide a label input information
http://v4-alpha.getbootstrap.com/components/input-group/#accessibility

**Comment by patrickhlauke** [comment]
on the last point about `aria-describedby` and `placeholder`: while not ideal, it is my understanding that on inputs at least the latter DOES act as a "last resort" way of providing a label via the accessibility API if none of the better ways are present (see http://rawgit.com/w3c/html-api-map/master/index.html#accessible-name-and-description-calculation). so based on that, the current example code is not exactly elegant, but "valid" (/cc @stevefaulkner)

having said that, adding an explicit `aria-label` seems the cleaner option for those inputs. and removing `aria-describedby` (as it's clearly wrong) and `placeholder` (as it's contentious) from the advice section makes sense.

**Comment by patrickhlauke** [comment]
xref https://github.com/w3c/html-aam/issues/84 which resulted in related bugs being filed against iOS/VoiceOver and Android/Chrome/Talkback
